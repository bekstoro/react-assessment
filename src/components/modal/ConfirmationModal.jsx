import React from 'react';
import PropTypes from 'prop-types';

import Portal from '@components/portal/Portal';
import {
  ModalButtons,
  ModalCancellationButton,
  ModalConfirmationButton,
  ModalText,
  ModalTitle,
  RedIcon,
  SmallModal,
} from '@components/styled-components/Modal';

export default function ConfirmationModal({
  cancellationButtonText,
  confirmationButtonText,
  isOpen,
  message,
  title,
  closeModal,
  confirmModal,
}) {
  if (!isOpen) return null;

  return (
    <Portal closeModal={closeModal}>
      <SmallModal data-testid="confirmation-modal">
        <ModalTitle>
          <RedIcon className="material-icons">warning</RedIcon>
          {title}
        </ModalTitle>
        <ModalText>{message}</ModalText>
        <ModalButtons>
          <ModalConfirmationButton
            onClick={() => {
              confirmModal();
              closeModal();
            }}
          >
            {confirmationButtonText}
          </ModalConfirmationButton>
          <ModalCancellationButton onClick={closeModal}>
            {cancellationButtonText}
          </ModalCancellationButton>
        </ModalButtons>
      </SmallModal>
    </Portal>
  );
}

ConfirmationModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  cancellationButtonText: PropTypes.string.isRequired,
  confirmationButtonText: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  confirmModal: PropTypes.func.isRequired,
};
