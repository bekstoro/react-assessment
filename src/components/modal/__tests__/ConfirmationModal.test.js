import React from 'react';
import { render } from '@testing-library/react';

import ConfirmationModal from '@components/modal/ConfirmationModal';

describe('ConfirmationModal', () => {
  beforeEach(() => {
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'app');
    document.body.appendChild(modalRoot);
  });

  it('ConfirmationModal renders correctly', () => {
    const { container } = render(
      <ConfirmationModal
        closeModal={jest.fn()}
        cancellationButtonText=""
        confirmationButtonText=""
        confirmModal={jest.fn()}
        isOpen
        message=""
        title=""
      />
    );
    expect(container).toMatchSnapshot();
  });
});
