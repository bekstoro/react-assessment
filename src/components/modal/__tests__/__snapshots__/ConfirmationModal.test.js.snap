// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal ConfirmationModal renders correctly 1`] = `<div />`;
